src/devdox_ai_git/utils/repository_url_parser.py

```
@@ -3,13 +3,20 @@
 import re
 from dataclasses import dataclass
 from urllib.parse import urlparse
+
 from devdox_ai_git.exceptions.base_exceptions import DevDoxGitException
 from devdox_ai_git.exceptions.exception_constants import (
     MISSING_NAMESPACE__REPO,
     UNRECOGNIZED_GIT_FORMAT,
 )
 from devdox_ai_git.schema.repo import GitHosting
 
+from devdox_ai_git.exceptions.base_exceptions import DevDoxGitException
```

Ruff (0.13.2)

Redefinition of unused `DevDoxGitException` from line 7

Remove definition: `DevDoxGitException`

(F811)